src/components/views/elements/FacePile.tsx

```diff
@@ -22,21 +22,62 @@ import { sortBy } from "lodash";
 import MemberAvatar from "../avatars/MemberAvatar";
 import { _t } from "../../../languageHandler";
 import DMRoomMap from "../../../utils/DMRoomMap";
+import TooltipTarget from "../elements/TooltipTarget";
 import TextWithTooltip from "../elements/TextWithTooltip";
 import { useRoomMembers } from "../../../hooks/useRoomMembers";
 import MatrixClientContext from "../../../contexts/MatrixClientContext";
 
+interface IProps extends HTMLAttributes<HTMLSpanElement> {
+    members: RoomMember[];
+    faceSize: number;
+    overflow: boolean;
+    tooltip?: ReactNode;
+    children?: ReactNode;
+}
+
+const FacePile: FC<IProps> = ({ members, faceSize, overflow, tooltip, children, ...props }) => {
+    const faces = members.map(
+        tooltip ?
+            m => <MemberAvatar key={m.userId} member={m} width={faceSize} height={faceSize} /> :
+            m => <TooltipTarget key={m.userId} label={m.name}>
+                <MemberAvatar member={m} width={faceSize} height={faceSize} />
+            </TooltipTarget>,
+    );
+
+    const pileContents = <>
+        { overflow ? <span className="mx_FacePile_more" /> : null }
+        { faces }
+    </>;
+
+    return <div {...props} className="mx_FacePile">
+        { tooltip ? (
+            <TextWithTooltip class="mx_FacePile_faces" tooltip={tooltip} tooltipProps={{ yOffset: 32 }}>
+                { pileContents }
+            </TextWithTooltip>
+        ) : (
+            <div className="mx_FacePile_faces">
+                { pileContents }
+            </div>
+        ) }
+        { children }
+    </div>;
+};
+
+export default FacePile;
+
 const DEFAULT_NUM_FACES = 5;
 
-interface IProps extends HTMLAttributes<HTMLSpanElement> {
+const isKnownMember = (member: RoomMember) => !!DMRoomMap.shared().getDMRoomsForUserId(member.userId)?.length;
+
+interface IRoomProps extends HTMLAttributes<HTMLSpanElement> {
     room: Room;
     onlyKnownUsers?: boolean;
     numShown?: number;
 }
 
-const isKnownMember = (member: RoomMember) => !!DMRoomMap.shared().getDMRoomsForUserId(member.userId)?.length;
-
-const FacePile: FC<IProps> = ({ room, onlyKnownUsers = true, numShown = DEFAULT_NUM_FACES, ...props }) => {
+export const RoomFacePile: FC<IRoomProps> = (
```

Comment:
file should be renamed to `RoomFacePile` to reflect new component name, and update any existing imports.

Reply:
FacePile is still there as the default export in that file, and RoomFacePile is just a shortcut for it, so I think this should keep its name

Reply:
this starts to get very tricky under the proposed code style - can we move `RoomFacePile` out to its own file then?